### DISCUSSION TITLE
How is the webhook setup in a template from Sanity team

### DISCUSSION THREAD

**Discussion by szymonhernik** [sanity-io/sanity#5235] [Requests for comment (RFC)] [answered]
Hello,

I've set up a template from Sanity (https://github.com/sanity-io/template-nextjs-personal-website). The template works great, thank you (!) ... but I'm wondering **how is the webhook set**? 

I expected it to find it in "GROQ-powered Webhooks" in Sanity panel but it's not there, **yet what it is responsible for (revalidation) works perfectly**. 

Could someone explain how is it able to update server components by triggering the api/revalidate route?

Thanks for clarifying.

**Comment by Evavic44** [comment, marked as answer]
Hey, @szymonhernik I wrote an article explaining these concept. Do give it a read: https://victoreke.com/blog/sanity-webhooks-and-on-demand-revalidation-in-nextjs

**Reply by szymonhernik** [reply]
Thank you, both the article and nextjs documentation helped a lot to wrap my head around it.
